plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -918,3 +918,41 @@ func (checker *PodAffinityChecker) NodeMatchPodAffinityAntiAffinity(pod *api.Pod
 	}
 	return true
 }
+
+// Determine if a pod is scheduled with best-effort QoS
+// Definition of best-effort pod by vishh: https://github.com/kubernetes/kubernetes/pull/14943
+func isPodBestEffort(pod *api.Pod) bool {
+	// A Pod is BestEffort iff requests & limits are not specified
+	// for any resource across all containers.
+	for _, container := range pod.Spec.Containers {
```

Comment:
you can just use: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/qos/util/qos.go#L52